src/ppx/xr_component.cpp

```diff
@@ -856,40 +847,72 @@ glm::mat4 XrComponent::GetProjectionMatrixForCurrentViewAndSetFrustumPlanes(floa
     mNearPlaneForFrame = nearZ;
     mFarPlaneForFrame  = farZ;
 
-    const float tan_left  = tanf(fov.angleLeft);
-    const float tan_right = tanf(fov.angleRight);
-
-    const float tan_down = tanf(fov.angleDown);
-    const float tan_up   = tanf(fov.angleUp);
-
-    const float tan_width  = tan_right - tan_left;
-    const float tan_height = (tan_up - tan_down);
-
-    const float a00 = 2 / tan_width;
-    const float a11 = 2 / tan_height;
-
-    const float a20 = (tan_right + tan_left) / tan_width;
-    const float a21 = (tan_up + tan_down) / tan_height;
-    const float a22 = -farZ / (farZ - nearZ);
+    for (XrCamera& camera : mCameras) {
+        camera.SetFrustumPlanes(nearZ, farZ);
+    }
+}
 
-    const float a32 = -(farZ * nearZ) / (farZ - nearZ);
+// -------------------------------------------------------------------------------------------------
+// XrCamera
+// -------------------------------------------------------------------------------------------------
 
-    // clang-format off
-    const float mat[16] = {
-        a00,    0,      0,      0,
-        0,      a11,    0,      0,
-        a20,    a21,    a22,    -1,
-        0,      0,      a32,    0,
-    };
-    // clang-format on
+void XrCamera::UpdateView(const XrView& view)
+{
+    mView = view;
+    XrCamera::UpdateCamera();
+}
 
-    return glm::make_mat4(mat);
+void XrCamera::SetFrustumPlanes(float nearZ, float farZ)
+{
+    if (mNearClip == nearZ && mFarClip == farZ) {
+        return;
+    }
+    mNearClip = nearZ;
+    mFarClip  = farZ;
+    UpdateCamera();
 }
 
-XrPosef XrComponent::GetPoseForCurrentView() const
+void XrCamera::UpdateCamera()
 {
-    PPX_ASSERT_MSG((mCurrentViewIndex < mViews.size()), "Invalid view index!");
-    return mViews[mCurrentViewIndex].pose;
+    const float3 forward = float3(0, 0, -1);
+    const float3 up      = float3(0, 1, 0);
+    // Given:
+    //   - mPixelAligned
+    //   - mNearClip, mFarClip
+    //   - view.fov.{angleLeft,angleRight,angleDown,angleUp}
+    //   - view.pose.{orientation,position}
+    // Copied:
+    //   - mEyePosition = view.pose.position
+    // Solve for:
+    //   - mViewDirection
+    //   - mTarget
+    //   - mWorldUp
+    //   - all matrices (using LookAt)
+
+    // Calculate projection matrix
+    {
+        const XrFovf& fov = mView.fov;
+
+        const float left  = tanf(fov.angleLeft) * mNearClip;
+        const float right = tanf(fov.angleRight) * mNearClip;
+
+        const float down = tanf(fov.angleDown) * mNearClip;
+        const float up   = tanf(fov.angleUp) * mNearClip;
+
+        // Not sure why it is frustumRH_ZO
```

Comment:
I'm still trying to figure out if `glm::frustum` would work or do I need this specific variant, it was hardcoded matrix coefficient.

Reply:
My understanding is as long as we don't define `GLM_FORCE_LEFT_HANDED` (which doesn't seem the case), GLM defaults to `RH` so this part shouldn't change.
However, since we use [0;1] for depth, the `math_config.h` defines GLM_FORCE_DEPTH_ZERO_TO_ONE, so I'd say in the current config, those are equivalent:

```
glm::frustum
glm::frustumZO
glm::frustumRH_ZO
```

I'd be in favor of using `glm::frustum` as it would allow changing the defines in `ppx/math_config.h` and keep the correct behavior, with the caveats that we should define the unit vectors `up`/`right`/`forward` and the `near`/`far` distances someplace unique.